WcaOnRails/spec/requests/api_competitions_spec.rb

```diff
@@ -7,6 +7,64 @@
     let(:competition) { FactoryBot.create(:competition, :with_delegate, :with_organizer, :visible) }
 
     describe "website user (cookies based)" do
+      let(:competition_events_add_event) {[
```

Comment:
Thanks for pulling this out! One problem with pulling this variable out like this is that it makes it hard to see *which* events are getting added. I would suggest that instead, you create a `def create_wcif_events` method at the bottom of this test that does something like:

```ruby
def create_wcif_events(event_ids)
  event_ids.map do |event_id|
    {
      id: event_id,
      rounds: [
        {
          id: "#{event_id}-1",
          format: "a",
          timeLimit: nil,
          cutoff: nil,
          advancementCondition: nil,
          scrambleGroupCount: 1,
        }
      ],
    }
  end
end
```

Then, tests can use this by doing something like `create_wcif_events(%w(333 333oh 222))`, which I think will make it more obvious what's going on in the tests. This also means you can get rid of the `competition_events_remove_event` variable.

Reply:
Dang, that's much nicer and easier to read. I will implement this now. Thank you!